### PR TITLE
fix(scripts): handle null response in RPC health check

### LINKED ISSUE #69: RPC ID Mismatch Detected
RPC Health Check (FULL mode)
============================================================

Fetching auth tokens...
Auth OK (CSRF token length: 42)

Creating temp resources for full testing...
OK       CREATE_NOTEBOOK - temp notebook created
OK       ADD_SOURCE - temp source added
OK       ADD_SOURCE_FILE - file source registered
OK       START_FAST_RESEARCH - research started
OK       CREATE_NOTE - temp note created
OK       CREATE_ARTIFACT - flashcard generation triggered
  Artifact ready after 2s polling

Checking 39 RPC methods...
============================================================
OK       LIST_NOTEBOOKS (wXbhsf)
SKIP     CREATE_NOTEBOOK (CCqFvf) - Tested in setup/cleanup phases
OK       GET_NOTEBOOK (rLM1Ne)
OK       RENAME_NOTEBOOK (s0tc2d)
SKIP     DELETE_NOTEBOOK (WWINqb) - Tested in setup/cleanup phases
SKIP     ADD_SOURCE (izAoDd) - Tested in setup/cleanup phases
SKIP     ADD_SOURCE_FILE (o4cbdc) - Tested in setup/cleanup phases
SKIP     DELETE_SOURCE (tGMBJ) - Tested in setup/cleanup phases
OK       GET_SOURCE (hizoJc)
OK       REFRESH_SOURCE (FLmJqe)
OK       CHECK_SOURCE_FRESHNESS (yR9Yof)
OK       UPDATE_SOURCE (b7Wfje)
SKIP     DISCOVER_SOURCES (qXyaNe) - Method always skipped (complex setup or quota)
OK       SUMMARIZE (VfAZjd)
OK       GET_SOURCE_GUIDE (tr032e)

Testing DELETE operations during cleanup...
OK       DELETE_NOTE - temp note deleted
OK       DELETE_SOURCE - temp source deleted
OK       DELETE_ARTIFACT - temp artifact deleted
OK       DELETE_NOTEBOOK - temp notebook deleted

### PR DESCRIPTION
**Pull request by teng-lin**
## Summary
- Add null check for `response_text` in `make_rpc_call()` to prevent `AttributeError` when the server returns an empty/null response

## Problem
The RPC health check script crashed with:
```
AttributeError: 'NoneType' object has no attribute 'startswith'
  File ".../check_rpc_health.py", line 264, in make_rpc_call
    cleaned = strip_anti_xssi(response_text)
```

This caused the CI to fail and incorrectly create issue #69 with title "RPC ID Mismatch Detected" when there was no actual RPC ID mismatch - just a missing null check.

## Fix
Added a defensive check for `response_text is None` before calling `strip_anti_xssi()`.

## Test plan
- [ ] CI passes
- [ ] Manually verified the fix handles the edge case

Fixes #69

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: scripts/check_rpc_health.py [hunk 1]
```diff
@@ -259,6 +259,8 @@ async def make_rpc_call(
     response_text, error = await make_rpc_request(client, auth, method, params, source_path)
     if error:
         return [], error
+    if response_text is None:
+        return [], "Empty response from server"
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply scripts/check_rpc_health.py]: Addressed in commit b65cbb9: added the same null check to `test_rpc_method_with_data` function. Good catch!
